### PR TITLE
[PoC] Introduce DearImGui

### PR DESCRIPTION
**Pull request by alxbilger**
The idea is to use DearImGui (https://github.com/ocornut/imgui) as a GUI for the SofaGLFW application.
This is a proof of concept that still needs some love, but the prototype is working.
It is much faster to render than Qt, and much faster to develop than Qt.

The GUI is not compiled by default. `SOFA_BUILD_IMGUI` must be set to `ON`.

DearImGui is fetched: no need to install anything.

The following features are implemented:

- Play/pause simulation
- Reset simulation
- Show performances information
- Show scene graph
- Show Datas of a component in the scene graph
- Change visual style
- Show loaded plugins list
- Show loaded components list

Windows support docking and multiviewport (from the docking branch https://github.com/ocornut/imgui/tree/docking). I noted performances drops when the window is out of the main window.

https://user-images.githubusercontent.com/10572752/145219593-476e77af-f27f-48f6-a3cc-35dcb223f6e4.mp4

SofaGLFW 2021-12-08 14-37-27_Moment

### CODE REVIEW

File: lib/src/SofaGLFW/SofaGLFWBaseGUI.cpp [hunk 1]
```diff
@@ -113,8 +114,8 @@ sofa::component::visualmodel::BaseCamera::SPtr SofaGLFWBaseGUI::findCamera(sofa:
 
 bool SofaGLFWBaseGUI::createWindow(int width, int height, const char* title, bool fullscreenAtStartup)
 {
-    //glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 2);
-    //glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 0);
+    imgui::imguiInit();
```
Comment (fredroy) [inline lib/src/SofaGLFW/SofaGLFWBaseGUI.cpp:117]: I suppose this will make a compilation error if imgui is not enabled ?

Comment (alxbilger) [reply lib/src/SofaGLFW/SofaGLFWBaseGUI.cpp:117]: No. In the current proposition, there is a #ifdef around the content of the function. In a next iteration, we could put it around the function call instead.

**Comment by alxbilger** [comment]
image

The log window

**Comment by alxbilger** [comment]
New features:

1. File browser to open any simulation file
2. The whole viewport is a dock so that any window can be docked on it.

image

**Review by fredroy — APPROVED** [review]

**Pull request merged**